sdk/python/feast/inference.py

```diff
@@ -111,6 +111,7 @@ def update_data_sources_with_inferred_event_timestamp_col(
             assert (
                 isinstance(data_source, FileSource)
                 or isinstance(data_source, BigQuerySource)
+                or isinstance(data_source, RedshiftSource)
```

Comment:
would you mind also adding the ""SparkSource" == data_source.__class__.__name__" line here so our spark source passes this too? (and updating the PR title)

Reply:
is there a reason we explicitly check the class name rather than doing `isinstance(data_source, SparkSource)`?

Reply:
yeah main reason is Spark is right now still a contrib plugin that requires an extra. If we use the explicit check, then that will break other users who don't install spark.